docs/index.rst

```diff
@@ -4,4 +4,315 @@ JDBC Connector
+tolerance, work with the JDBC connector. You can restart and kill the processes and they will
+pick up where they left off, copying only new data (as defined by the ``mode`` setting).
+
+Features
+--------
+
+The JDBC connector supports copying tables with a variety of JDBC data types, adding and removing
+tables from the database dynamically, whitelists and blacklists, varying polling intervals, and
+other settings. However, the most important features for most users are the settings controlling
+how data is incrementally copied from the database.
+
+Mapping to Kafka Connect Partitioned Streams & Offsets
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Kafka Connect uses `partitioned streams` as its core data model. In order to map to Kafka Connect
+records, the JDBC connector must model its data as a set of partitions, each with an ordered
+sequence of updates with unique offsets. In the case of a process restart or failure, these
+offsets are used to resume reading from the last position that the system can guarantee has been
+processed.
+
+To map to this model, the JDBC connector must map partitions and offsets. Partitions is
+handled on a per-table basis: each table represents a single partition and every insertion or
+update of a row corresponds to a single record in that partition. As a special case, custom
+queries (which disable table-based processing) are treated as a single-partition stream.
+
+Offsets now must be defined for each table's stream. The JDBC connector supports multiple options
+depending on the format of data within your tables. However, at their core, all settings share a
+common approach. A column or set of columns is used to define and order offsets. A simple example
+uses only unique, auto-incrementing IDs in a table where rows are only inserted, never updated.
+In this mode, offsets are a single integer ID and are naturally ordered. Querying for
+only new data is trivially handled by checking for IDs with larger values than the
+largest seen so far. "Rewinding" upon failure simply requires looking up the largest ID
+previously seen and starting queries with that value.
+
+However, many tables cannot be guaranteed to only have insertions and have a unique,
+auto-incrementing ID. In order to support a variety of tables with different properties, multiple
+different approaches to defining offsets & queries for new or updated rows are supported.
+
+Incremental Query Modes
+~~~~~~~~~~~~~~~~~~~~~~~
+
+* **Increasing Column**: In this mode, a single column is specified and contains a unique ID for
+  each row which increases with each insertion. Although it could be implemented differently, this
+  type of column is most frequently implemented as an ``AUTOINCREMENT`` column in a table where rows
+  are only inserted, but not updated. (Note that updates will *not* be handled properly in this
+  mode because they cannot be detected once the row has been processed for the first time.)
```

Comment:
I'd add something to the effect of "this is a good fit when streaming fact tables in a data warehouse, since those are typically insert-only".